src/synthesizer/csharp/mod.rs

```diff
@@ -0,0 +1,668 @@
+            trailing: Some("}".into()),
+            trailing_newline: true,
+        });
+
+        // Properties for each output
+        for output in &ir.outputs {
+            if let Some(description) = &output.description {
+                stack_class.line("/// <summary>");
+                for description_line in description.split('\n') {
+                    stack_class.line(format!("/// {description_line}"));
+                }
+                stack_class.line("/// </summary>");
+            }
+            stack_class.line(format!("public object {} {{ get; }} ", output.name));
+            stack_class.newline();
+        }
+
+        // Constructor
+        let ctor = stack_class.indent_with_options(IndentOptions {
+            indent: INDENT,
+            leading: Some(format!(
+                "public {}(Construct scope, string id, {}Props props = null) : base(scope, id, props)\n{{",
+                stack_name,
+                stack_name
+            ).into()),
+            trailing: Some("}".into()),
+            trailing_newline: true,
+        });
+
+        // Default values for input props
+        let have_default_or_special_type_params = &ir
+            .constructor
+            .inputs
+            .iter()
+            .filter(|p| p.constructor_type.contains("AWS::") || p.default_value.is_some())
+            .collect::<Vec<&ConstructorParameter>>();
+        if !have_default_or_special_type_params.is_empty() {
+            ctor.line("// Applying default props");
+            for param in have_default_or_special_type_params {
+                let name = pascal_case(&param.name);
+                // example: AWS::EC2::Image::Id, List<AWS::EC2::VPC::Id>, AWS::SSM::Parameter::Value<List<String>>
+                if param.constructor_type.contains("AWS::") {
+                    let value_as = match &param.constructor_type {
+                        t if t.contains("List") => "ValueAsList",
+                        _ => "ValueAsString",
+                    };
+                    let cfn_param = ctor.indent_with_options(IndentOptions {
+                        indent: INDENT,
+                        leading: Some(
+                            format!(
+                                "props.{name} = new CfnParameter(this, \"{name}\", new CfnParameterProps\n{{")
+                            .into(),
+                        ),
+                        trailing: Some(format!("}}).{value_as};").into()),
+                        trailing_newline: true,
+                    });
+                    cfn_param.line(format!("Type = \"{}\",", param.constructor_type));
+                    let cfn_param_default_pre = match &param.constructor_type {
+                        t if t.contains("List") => "string.Join(\",\", ",
+                        _ => "",
+                    };
+                    let cfn_param_default_post = match &param.constructor_type {
+                        t if t.contains("List") => ")",
```

Comment:
Since names can be reused in Rust, we can def use shorter ones and reuse var names.

Thoughts on alternative names:
```
list_optional_prefix / list_optional_suffix
list_preamble / list_postamble
```

I'm also not the greatest wordsmith 😂 -- but when reading down below, I was like `that's verbose and I don't know what it is when reading it` which bugged me (and why I like adding the type of content in the name.

Reply:
Thanks for this feedback, I applied your first suggestion. Let me know if I missed part of this. I'm not sure how to apply reusing var names in this scenario.